apps/src/lib/kits/maker/PlaygroundComponents.js

```diff
@@ -54,6 +54,64 @@ export function initializeCircuitPlaygroundComponents(board) {
 }
 
 /**
+ * De-initializes any Johnny-Five components that might have been created
+ * by createCircuitPlaygroundComponents
+ * @param {Object} components - map of components, as originally returned by
+ *   createCircuitPlaygroundComponents.  This object will be mutated: Destroyed
+ *   components will be removed. Additional members of this object will be
+ *   ignored.
+ */
+export function destroyCircuitPlaygroundComponents(components) {
+  if (components.colorLeds) {
+    components.colorLeds.forEach(led => led.stop());
+  }
+  delete components.colorLeds;
```

Comment:
Deleting these components from the provided object isn't especially useful in production code (we throw that object away a moment later anyway) but it has some really nice effects on testability - we've now got [a test that will fail if a new component is added to `createCircuitPlaygroundComponents` without matching cleanup code in `destroyCircuitPlaygroundComponents`.](https://github.com/code-dot-org/code-dot-org/pull/13638/files#diff-ca2026850a1bf45fc49a50c5fc166c80R372)